src/js/form-builder.js

```diff
@@ -98,7 +96,7 @@ const FormBuilder = function(opts, element) {
     }
     let icon = custom.icon || controlClass.icon(type);
     let label = custom.label || controlClass.label(type);
-    let iconClassName = !icon ? custom.iconClassName || `icon-${type}` : '';
+    let iconClassName = !icon ? custom.iconClassName || `icon-${type.replace(/-[\d]{4}$/, '')}` : '';
```

Comment:
Yep I guess either way works. My implementation above removes ignores the '-1234' by just using the original type defined in `opts.fields`. So
```
var fields [
  {
    type => 'text',
      ...
  }
];
```

... would then output `icon-text` for that custom field.

Either way should have the same outcome so up to you which you'd prefer.